### PR TITLE
Zubosem

### PR DESCRIPTION
**Pull request by zubosem**
- Fix payments feature
- Fix group's tip feature
- Fix minimal amount to 0.01 in validator rule of WalletTransaction
- Fix bug when bot make duplicate messages
- Redirect tip's payment process to WalletController
- Write some Wallet and WalletTransaction tests
- Up friendsofphp/php-cs-fixer version in composer.
- Add rule single_space_around_construct to php-cs-fixer config
- Add some fixes to INSTALL and CONTRIBUTING docs

### CODE REVIEW

File: modules/bot/components/api/Types/Update.php [hunk 1]
```diff
@@ -70,7 +70,13 @@ public function map($data)
 
     public function setPrivateMessageFromState(UserState $state)
     {
-        $privateMessageIds = json_decode($state->getIntermediateField('private_message_ids', json_encode([])));
+        $privateMessageIds = $state->getIntermediateField('private_message_ids', []);
+
+        // fix previous invalid saving
+        if(is_string($privateMessageIds))
```
Comment (grandmotivator) [inline modules/bot/components/api/Types/Update.php:76]: Fix code style

File: modules/bot/controllers/privates/WalletController.php [hunk 2]
```diff
@@ -307,7 +343,7 @@ public function actionSendTransaction($id = null)
         return $this->getResponseBuilder()
             ->editMessageTextOrSendMessage(
                 $this->render('send-transaction'),
-                $buttons
+                $buttons,
```
Comment (grandmotivator) [inline modules/bot/controllers/privates/WalletController.php:346]: It is error

File: modules/bot/controllers/privates/WalletController.php [hunk 3]
```diff
@@ -325,28 +361,29 @@ public function actionSetToUser()
...
-        if ($text = $this->getMessage()->getText()) {
-            if (preg_match('/(?:^@(?:[A-Za-z0-9][_]{0,1})*[A-Za-z0-9]+)/i', $text, $matches)) {
-                $username = ltrim($matches[0], '@');
-            }
+        $text = $this->getMessage()->getText();
+
+        if (preg_match('/(?:^@(?:[A-Za-z0-9][_]{0,1})*[A-Za-z0-9]+)/i', $text, $matches)) {
+            $username = ltrim($matches[0], '@');
+            $toBotUser = User::findOne(['provider_user_name' => $username]);
+        } elseif (preg_match('/^\d+$/', $text)) {
```
Comment (grandmotivator) [inline modules/bot/controllers/privates/WalletController.php:369]: It is not required to add the ability to transfer for the user id.

**Comment by zubosem** [comment]
- fix composer
- fix syntax
- remove unnecessary operation

**Pull request merged**